### PR TITLE
fix floatThead firefox header misalignment + refactor some related code

### PR DESCRIPTION
**Pull request by mkoryak**
main things in this pr:
- fix that ff header misalignment 'bug' by changing table `width:100%` to `min-width:100%`
- always show percentages while scrolling
- refactor some js code make calling floatThead less weird

I tried to not go too crazy with refactoring, but you might feel I did more than this PR would warrant - let me know and ill make 2.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ es*/compilers/*
 .DS_Store
 .ruby-version
 _site/
+.idea/
```
Comment (kangax) [inline .gitignore:6]: Out of curiosity, what is this for?

File: master.js [hunk 2]
```diff
@@ -44,67 +44,43 @@ $(function() {
...
-    };
-    return fn;
-  })(table.floatThead);
-
-  // Set up the Show Obsolete checkbox
-  $('#show-obsolete, #show-unstable').on('click', function() {
-    setTimeout(function() {
+  if (false) { //set to true to disable floating headers
+    // to fully remove, search for floatThead and remove all
+    $.fn.floatThead = function(){ return this };
```
Comment (kangax) [inline master.js:50]: Nice :)

File: master.js [hunk 3]
```diff
@@ -446,7 +424,7 @@ $(function() {
                           : 'engines';
 
     // First, hide the platformtype bar if we're sorting by features.
-    $('.platformtype')[(sortByFeatures || sortByFlaggedFeatures) ? 'hide' : 'show']();
+    $('body')[(noPlatformtypeBar || sortByFeatures || sortByFlaggedFeatures) ? 'addClass' : 'removeClass']('hide-platformtype');
```
Comment (kangax) [inline master.js:427]: I know we already had it like this but should probably use `toggleClass` here at some point

**Comment by kangax** [comment]
I love the animated transition on percentages when you scroll! Feels nice, thanks @mkoryak

File: .gitignore (hunk 1 above)
Comment (ljharb) [inline .gitignore:6]: blocks IntelliJ files - imo it should be in the user's global gitignore rather than an individual project's, but it can't hurt.

Comment (kangax) [inline .gitignore:6]: Gotcha, thanks

**Comment by mkoryak** [comment]
no problem, thanks for using my plugin, though you had no choice, since all the other ones suck ;)

File: master.js (hunk 3 above)
Comment (mkoryak) [inline master.js:427]: ah, I see, toggle class can take a `state`, that would be cleaner

**Pull request merged**